src/components/common/DocumentDelivery.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <v-card flat class="px-5 py-6" id="document-delivery">
+    <v-row class="py-3" no-gutters>
+      <v-col :cols="3">
+        <label class="font-weight-bold">Registered Office</label>
+      </v-col>
+      <v-col :cols="9">
+        <v-card-text id="office-email" class="pa-0">{{getBusinessContact.email || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+    <v-row class="py-3" no-gutters>
+      <v-col :cols="3">
+        <label class="font-weight-bold" :class="{ 'error-text': invalidSection }">Completing Party</label>
+      </v-col>
+      <v-col cols="9" class="pr-1" v-if="editableCompletingParty">
+        <v-text-field
+          v-model="optionalEmail"
+          id="optionalEmail"
+          class="text-input-field"
+          filled
+          label="Client Email Address (Optional)"
+          hint="Example: name@email.com"
+          persistent-hint
+          validate-on-blur
+          :rules="entityEmailRules"
+        />
+      </v-col>
+      <v-col :cols="9" v-else>
+        <v-card-text id="completing-party-email" class="pa-0">{{getUserEmail || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+    <v-row class="py-3" no-gutters v-if="showCustodianEmail">
+      <v-col :cols="3">
+        <label class="font-weight-bold">Custodian of Records</label>
+      </v-col>
+      <v-col :cols="9">
+        <v-card-text id="custodian-email" class="pa-0">{{getCustodianEmail || 'Not entered'}}</v-card-text>
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF, BusinessContactIF, DocumentDeliveryIF } from '@/interfaces'
+
+@Component({})
+export default class DocumentDelivery extends Vue {
+  @Getter getBusinessContact!: BusinessContactIF
+  @Getter getCustodianEmail!: string
+  @Getter getDocumentDelivery!: DocumentDeliveryIF
+  @Getter getUserEmail!: string
+
+  // Global actions
+  @Action setDocumentOptionalEmail!: ActionBindingIF
+  @Action setDocumentOptionalEmailValidity!: ActionBindingIF
```

Comment:
Because of these getters and actions, this component can never be changed into a shared component (since it uses store).

The architecture I've tried to set for this app is that high-level components can read/write to store, but low-level components use props and events only.

Is this component a high-level component? It might be. I just want to make a conscious decision that it is.

Reply:
Changed it

Reply:
Oh wow, that made a lot of code a lot cleaner!
Closing this conversation.